sharedbuffers/mapped_struct.py

```diff
@@ -131,7 +131,7 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
                 return tup.pack_into(tup, buf, offs, idmap, implicit_offs)
         else:
             # Same representation as a tuple of items, only backed in-memory by a frozenset
-            tup = mapped_tuple(obj)
+            tup = mapped_tuple(obj if sort is False else sorted(obj))
```

Comment:
Sorting should be unconditional if feasible. The whole point is that we should be able to switch from regular frozensets to proxied ones without rebuilding the underlying representation (they should be compatible), and an unsorted frozenset wouldn't be compatible.

That said, you should try to make sorting as efficient as possible. Using python's "sorted" wouldn't achieve that. `mapped_tuple` has optimized paths for packing numpy arrays IIRC, sorted would break that. And numpy arrays can be sorted more efficiently with numpy itself.

Reply:
The input for frozenset construction can be an arbitrary sequence, so I'm not sure if we can sort unconditionally. I can apply a numpy-specific sorting since that's likely to be better.

Reply:
Items are hashable, so they're sortable by hash. We talked about it.